### PR TITLE
IDGEN-117 Idgen module initializing error after jetty server starts 

### PR DESCRIPTION
**Pull request by Am-Coder**
**Changes I have made :**
1.  Added uuid support for PostgreSQL
2.  Liquibase converts tinyint to smallint in case of PostgreSQL which does not behaves as boolean. For MySQL, tinyint(1) is equivalent to boolean but not for PostgreSQL. So I made the changes so that for PostgreSQL the column type is set to boolean and for other db's it is tinyint(1) . Also, **like** comparisons for boolean types gives error on PostgreSQL so made it an **equal** comparison in hibernate expression.

** Link to ticket **
https://issues.openmrs.org/browse/IDGEN-117

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/idgen/service/db/HibernateIdentifierSourceDAO.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ public IdentifierSource getIdentifierSource(Integer id) throws APIException {
 	public List<IdentifierSource> getAllIdentifierSources(boolean includeRetired) throws DAOException {
 		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(IdentifierSource.class);
 		if (!includeRetired) {
-			criteria.add(Expression.like("retired", false));
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/module/idgen/service/db/HibernateIdentifierSourceDAO.java:75]: Is this related?

File: api/src/main/resources/liquibase.xml [hunk 2]
```diff
@@ -34,7 +36,7 @@
 			<column name="date_changed" type="datetime">
 				<constraints nullable="true"/>
 			</column>
-			<column name="retired" type="tinyint(1)" defaultValue="0">
```
Comment (dkayiwa) [inline api/src/main/resources/liquibase.xml:37]: Why not BOOLEAN?

File: api/src/main/java/org/openmrs/module/idgen/service/db/HibernateIdentifierSourceDAO.java (hunk 1 above)
Comment (Am-Coder) [reply api/src/main/java/org/openmrs/module/idgen/service/db/HibernateIdentifierSourceDAO.java:75]: Yes with expression.like you get an error raised.

File: api/src/main/resources/liquibase.xml (hunk 2 above)
Comment (Am-Coder) [reply api/src/main/resources/liquibase.xml:37]: liquibase converts boolean to BIT(1) in MySQL ( as found [here](https://stackoverflow.com/questions/16890723/list-all-liquibase-sql-types) ) but according to MySQL standards Boolean is equivalent to Tinyint(1) ( as found [here](https://dev.mysql.com/doc/refman/8.0/en/integer-types.html) )

Comment (dkayiwa) [reply api/src/main/resources/liquibase.xml:37]: Doesn't openmrs-core already have `retired` fields as BOOLEAN?

Comment (Am-Coder) [reply api/src/main/resources/liquibase.xml:37]: @dkayiwa  you are right. Just found [this](https://github.com/openmrs/openmrs-core/blob/47463eb05661ec9f8323c96874fc8d6161551389/api/src/main/java/liquibase/ext/datatype/core/MySQLBooleanType.java) where we are explicitly converting Boolean to tinyint(1) in OpenMRS for MySQL. I'll try with boolean now.

Comment (Am-Coder) [reply api/src/main/resources/liquibase.xml:37]: @dkayiwa , I have made the changes .

File: api/src/main/resources/liquibase.xml [hunk 3]
```diff
@@ -362,6 +365,9 @@
 		<modifySql dbms="mssql">
 			<replace replace="name-of-uuid-function" with="NEWSEQUENTIALID()"/>
 		</modifySql>
+		<modifySql dbms="postgresql">
+			<replace replace="name-of-uuid-function" with="UUID()"/>
```
Comment (dkayiwa) [inline api/src/main/resources/liquibase.xml:369]: Don't we already have the UUID() function already taken care of in openmrs-core?

Comment (Am-Coder) [reply api/src/main/resources/liquibase.xml:369]: Yes, that is why we are using UUID() here and not uuid_generate_v1(). In openmrs-core we have created the wrapper function similar to MySQL for PostgreSQL which is UUID() that wraps around uuid_generate_v1().
